streaming/stream_blob.cc

```diff
@@ -0,0 +1,705 @@
+        stream_options.buffer_size = file_stream_buffer_size;
+        stream_options.write_behind = file_stream_write_behind;
+
+        auto& table = db.find_column_family(meta.table);
+        auto& sstm = table.get_sstables_manager();
+        auto sstable_sink = sstables::create_stream_sink(table.schema(), sstm, table.get_storage_options(), sstable_state(meta), meta.filename, meta.fops == file_ops::load_sstables);
+        auto out = co_await sstable_sink->output(foptions, stream_options);
+        co_return output_result{
+            [sstable_sink = std::move(sstable_sink), &meta, &db](store_result res) -> future<> {
+                if (res != store_result::ok) {
+                    co_await sstable_sink->abort();
+                    co_return;
+                }
+                auto sst = co_await sstable_sink->close_and_seal();
+                if (sst) {
+                    auto filename = sst->toc_filename();
+                    sst = {};
+                    co_await load_sstable_for_tablet(meta.ops_id, db, meta.table, sstable_state(meta), std::move(filename), meta.dst_shard_id);
+                }
+            },
+            std::move(out)
+        };
+    });
+}
+
+// Get a new sstable name using the new generation
+// For example:
+//     oldname: me-3ga1_0iiv_2e5uo2flv7lgdl2j0d-big-Index.db
+//      newgen:    3ga1_0iiv_3vj5c2flv7lgdl2j0d
+//     newname: me-3ga1_0iiv_3vj5c2flv7lgdl2j0d-big-Index.db
+static std::string get_sstable_name_with_generation(const file_stream_id& ops_id, const std::string& oldname, const std::string& newgen) {
+    std::string newname = oldname;
+    // The generation name starts after the first '-'.
+    auto it = newname.find("-");
+    if (it != std::string::npos) {
+        newname.replace(++it, newgen.size(), newgen);
+        return newname;
+    } else {
+        auto msg = fmt::format("fstream[{}] Failed to get sstable name for {} with generation {}", ops_id, oldname, newgen);
+        blogger.warn("{}", msg);
+        throw std::runtime_error(msg);
+    }
+}
+}
+
+template<> struct fmt::formatter<streaming::stream_blob_info> : fmt::ostream_formatter {};
+
+namespace streaming {
+
+// Send files in the files list to the nodes in targets list over network
+// Returns number of bytes sent over network
+future<size_t>
+tablet_stream_files(netw::messaging_service& ms, std::list<stream_blob_info> sources, std::vector<node_and_shard> targets, table_id table, file_stream_id ops_id, host2ip_t host2ip, service::frozen_topology_guard topo_guard, bool inject_errors) {
+    size_t ops_total_size = 0;
+    if (targets.empty()) {
+        co_return ops_total_size;
+    }
+    if (sources.empty()) {
+        co_return ops_total_size;
+    }
+
+    blogger.debug("fstream[{}] Master started sending n={}, sources={}, targets={}",
+            ops_id, sources.size(), sources, targets);
+
+    struct sink_and_source {
+        gms::inet_address node;
```

Comment:
Why is this no host_id based? Can we please do not introduce more ip based code into the master?